[2025-10-08T03:05:06.682+0000] {local_task_job_runner.py:120} INFO - ::group::Pre task execution logs
[2025-10-08T03:05:06.721+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: restaurant_analytics.run_spark_stream scheduled__2025-10-08T02:00:00+00:00 [queued]>
[2025-10-08T03:05:06.736+0000] {taskinstance.py:2076} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: restaurant_analytics.run_spark_stream scheduled__2025-10-08T02:00:00+00:00 [queued]>
[2025-10-08T03:05:06.747+0000] {taskinstance.py:2306} INFO - Starting attempt 2 of 2
[2025-10-08T03:05:06.775+0000] {taskinstance.py:2330} INFO - Executing <Task(BashOperator): run_spark_stream> on 2025-10-08 02:00:00+00:00
[2025-10-08T03:05:06.794+0000] {warnings.py:112} WARNING - /home/***/.local/lib/python3.12/site-packages/***/task/task_runner/standard_task_runner.py:62: DeprecationWarning: This process (pid=542) is multi-threaded, use of fork() may lead to deadlocks in the child.
  pid = os.fork()

[2025-10-08T03:05:06.799+0000] {standard_task_runner.py:64} INFO - Started process 546 to run task
[2025-10-08T03:05:06.802+0000] {standard_task_runner.py:90} INFO - Running: ['***', 'tasks', 'run', 'restaurant_analytics', 'run_spark_stream', 'scheduled__2025-10-08T02:00:00+00:00', '--job-id', '108', '--raw', '--subdir', 'DAGS_FOLDER/restaurant_dag.py', '--cfg-path', '/tmp/tmp835gsjwy']
[2025-10-08T03:05:06.807+0000] {standard_task_runner.py:91} INFO - Job 108: Subtask run_spark_stream
[2025-10-08T03:05:07.021+0000] {task_command.py:426} INFO - Running <TaskInstance: restaurant_analytics.run_spark_stream scheduled__2025-10-08T02:00:00+00:00 [running]> on host c34aac175a74
[2025-10-08T03:05:07.310+0000] {taskinstance.py:2648} INFO - Exporting env vars: AIRFLOW_CTX_DAG_EMAIL='quantruong1918@gmail.com' AIRFLOW_CTX_DAG_OWNER='***' AIRFLOW_CTX_DAG_ID='restaurant_analytics' AIRFLOW_CTX_TASK_ID='run_spark_stream' AIRFLOW_CTX_EXECUTION_DATE='2025-10-08T02:00:00+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='scheduled__2025-10-08T02:00:00+00:00'
[2025-10-08T03:05:07.315+0000] {taskinstance.py:430} INFO - ::endgroup::
[2025-10-08T03:05:07.319+0000] {subprocess.py:63} INFO - Tmp dir root location: /tmp
[2025-10-08T03:05:07.322+0000] {subprocess.py:75} INFO - Running command: ['/usr/bin/bash', '-c', 'spark-submit --master local[*] /opt/***/dags/spark_streaming.py']
[2025-10-08T03:05:07.352+0000] {subprocess.py:86} INFO - Output:
[2025-10-08T03:05:07.354+0000] {subprocess.py:93} INFO - /usr/bin/bash: line 1: spark-submit: command not found
[2025-10-08T03:05:07.356+0000] {subprocess.py:97} INFO - Command exited with return code 127
[2025-10-08T03:05:07.357+0000] {taskinstance.py:441} INFO - ::group::Post task execution logs
[2025-10-08T03:05:07.383+0000] {taskinstance.py:2905} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 465, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/taskinstance.py", line 432, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/models/baseoperator.py", line 401, in wrapper
    return func(self, *args, **kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/airflow/.local/lib/python3.12/site-packages/airflow/operators/bash.py", line 243, in execute
    raise AirflowException(
airflow.exceptions.AirflowException: Bash command failed. The command returned a non-zero exit code 127.
[2025-10-08T03:05:07.399+0000] {taskinstance.py:1206} INFO - Marking task as FAILED. dag_id=restaurant_analytics, task_id=run_spark_stream, run_id=scheduled__2025-10-08T02:00:00+00:00, execution_date=20251008T020000, start_date=20251008T030506, end_date=20251008T030507
[2025-10-08T03:05:07.453+0000] {standard_task_runner.py:110} ERROR - Failed to execute job 108 for task run_spark_stream (Bash command failed. The command returned a non-zero exit code 127.; 546)
[2025-10-08T03:05:07.503+0000] {local_task_job_runner.py:243} INFO - Task exited with return code 1
[2025-10-08T03:05:07.528+0000] {local_task_job_runner.py:222} INFO - ::endgroup::
